### PR TITLE
provide configurable HTTP handler wrapper for micro server

### PR DESCRIPTION
**Pull request by xh3b4sd**
This PR goes towards https://github.com/giantswarm/api/issues/292.

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -103,7 +106,10 @@ func DefaultConfig() Config {
 		TransactionResponder: responderService,
 
 		// Settings.
-		Endpoints:     nil,
+		Endpoints: nil,
+		HandlerWrapper: func(h http.Handler) http.Handler {
+			return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) { h.ServeHTTP(w, r) })
```
Comment (JosephSalisbury) [inline server/server.go]: why not just return `h`?

Comment (xh3b4sd) [inline server/server.go]: Because I am not worth being your colleague and you are a fucking genius Sir.

Comment (xh3b4sd) [inline server/server.go]: Done.

**Comment by xh3b4sd** [comment]
Please have another look good Sir.

File: server/server.go (hunk 1 above)
Comment (JosephSalisbury) [inline server/server.go]: <3

**Pull request merged**